comm.cpp

```diff
@@ -245,6 +245,17 @@ void Endpoint::print_statistics()
            _write_total);
 }
 
+int Endpoint::write_msg(const mavlink_message_t *msg)
+{
+    struct buffer buffer;
+    uint8_t data[MAVLINK_MAX_PACKET_LEN];
+
+    buffer.data = data;
+    buffer.len = mavlink_msg_to_send_buffer(data, msg);
```

Comment:
There is only one copy here from mavlink_message_t to buffer and buffer will be written to file descriptor.